Assets/Scripts/StandardSamples/UI/Common/SampleSelectableStateHandler.cs

```diff
@@ -0,0 +1,108 @@
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+* SOFTWARE.
+*/
+
+namespace PlayEveryWare.EpicOnlineServices.Samples
+{
+    using System.Collections;
+    using UnityEngine;
+    using UnityEngine.UI;
+    using UnityEngine.EventSystems;
+    using System;
+
+    public class SampleSelectableStateHandler : MonoBehaviour
+    {
+        public struct SampleInteractableNewState
+        {
+            public bool Interactable;
+            public string NewTooltipText;
+
+            public SampleInteractableNewState(bool shouldBeInteractable, string newTooltipText = "")
+            {
+                Interactable = shouldBeInteractable;
+                NewTooltipText = newTooltipText;
+            }
+        }
+
+        public delegate SampleInteractableNewState GetSampleInteractable();
+
+        private static event Action UpdateSelectableStates;
+
+        private GetSampleInteractable sampleInteractableFunction { get; set; }
+
+        [SerializeReference]
+        private UITooltip attachedTooltip;
+
+        [SerializeReference]
+        private Selectable attachedSelectable;
+
+        public void SetSampleInteractableAction(GetSampleInteractable inSampleInteractableFunction)
+        {
+            sampleInteractableFunction = inSampleInteractableFunction;
+            UpdateFromFunction();
+        }
+
+        private void OnEnable()
+        {
+            UpdateSelectableStates += UpdateFromFunction;
+        }
+
+        private void OnDisable()
+        {
+            UpdateSelectableStates -= UpdateFromFunction;
+        }
+
+        void UpdateFromFunction()
+        {
+            if (sampleInteractableFunction == null)
+            {
+                return;
+            }
+
+            SampleInteractableNewState newState = sampleInteractableFunction();
+
+            if (!string.IsNullOrEmpty(newState.NewTooltipText) && attachedTooltip != null)
+            {
+                if (newState.Interactable)
+                {
+                    // TODO: Restore the original tooltip text if it is interactable
+                    // At the moment, none of the buttons that this was applied to
+                    // had any tooltips to begin with
+                    attachedTooltip.Text = string.Empty;
+                }
+                else
+                {
+                    attachedTooltip.Text = newState.NewTooltipText;
+                }
+            }
+
+            if (attachedSelectable != null)
+            {
+                attachedSelectable.interactable = newState.Interactable;
+            }
+        }
+
+        public static void RaiseSampleSelectableStateChange()
```

Comment:
To me the word 'Raise' implies that an exception or signal is being 'raised'; perhaps we can change it to something like 'notify', 'invoke', 'dispatch'

Reply:
Renamed to NotifySampleSelectableUpdate.